src/test/plugins/shared_dictionaries_plugin_test.cpp

```diff
@@ -0,0 +1,265 @@
+    return sm.get_table("customer");
+  }
+
+  const SegmentChunkPair<pmr_string> _get_segment_chunk_pair() {
+    const auto chunk = _get_customer_table()->get_chunk(ChunkID{0});
+    const auto segment = chunk->get_segment(_market_column_id);
+    const auto dictionary_segment = std::dynamic_pointer_cast<DictionarySegment<pmr_string>>(segment);
+    return std::make_pair(dictionary_segment, chunk);
+  }
+
+  std::string _get_table_path(const std::string& table_name) { return _tables_path + table_name + _table_extension; }
+
+  static double _calc_jaccard_index(size_t union_size, size_t intersection_size) {
+    return SharedDictionariesColumnProcessor<int32_t>::_calc_jaccard_index(union_size, intersection_size);
+  }
+
+  static void _initialize_merge_plans(
+      SharedDictionariesColumnProcessor<pmr_string>& column_processor,
+      std::vector<std::shared_ptr<SharedDictionariesColumnProcessor<pmr_string>::MergePlan>>& merge_plans) {
+    column_processor._initialize_merge_plans(merge_plans);
+  }
+
+  static bool _should_merge(SharedDictionariesColumnProcessor<pmr_string>& column_processor, const double jaccard_index,
+                            const size_t current_dictionary_size, const size_t shared_dictionary_size,
+                            const std::vector<SegmentChunkPair<pmr_string>>& shared_segment_chunk_pairs) {
+    return column_processor._should_merge(jaccard_index, current_dictionary_size, shared_dictionary_size,
+                                          shared_segment_chunk_pairs);
+  }
+
+  static void _add_segment_chunk_pair(SharedDictionariesColumnProcessor<pmr_string>::MergePlan& merge_plan,
+                                      const SegmentChunkPair<pmr_string>& segment_chunk_pair, bool is_already_merged) {
+    SharedDictionariesColumnProcessor<pmr_string>::_add_segment_chunk_pair(merge_plan, segment_chunk_pair,
+                                                                           is_already_merged);
+  }
+};
+
+TEST_F(SharedDictionariesPluginTest, LoadUnloadPluginEmpty) {
+  auto& pm = Hyrise::get().plugin_manager;
+  pm.load_plugin(build_dylib_path("libhyriseSharedDictionariesPlugin"));
+  pm.unload_plugin("hyriseSharedDictionariesPlugin");
+}
+
+TEST_F(SharedDictionariesPluginTest, LoadUnloadPlugin) {
+  _add_all_test_tables();
+  auto& pm = Hyrise::get().plugin_manager;
+  pm.load_plugin(build_dylib_path("libhyriseSharedDictionariesPlugin"));
+  _validate_all_test_tables();
+  pm.unload_plugin("hyriseSharedDictionariesPlugin");
+}
+
+TEST_F(SharedDictionariesPluginTest, ReloadPlugin) {
+  _add_all_test_tables();
+  auto& pm = Hyrise::get().plugin_manager;
+
+  pm.load_plugin(build_dylib_path("libhyriseSharedDictionariesPlugin"));
+  _validate_all_test_tables();
+  pm.unload_plugin("hyriseSharedDictionariesPlugin");
+
+  pm.load_plugin(build_dylib_path("libhyriseSharedDictionariesPlugin"));
+  _validate_all_test_tables();
+  pm.unload_plugin("hyriseSharedDictionariesPlugin");
+}
```

Comment:
Since `LoadUnloadPlugin` is already a part of the `ReloadPlugin` test, consider removing the`LoadUnloadPlugin` test

Reply:
We decided to keep both tests as it is possible for one test to fail and the other not.
So if `LoadUnloadPlugin` is successful and `ReloadPlugin` is not, the user sees at a glance that the error is on the reload.